### PR TITLE
GroupedWithin cold emit fix 

### PR DESCRIPTION
**Pull request by Rattenkrieg**
Fix for #22406

### CODE REVIEW

**Comment by typesafehub-validator** [comment]
Hi @Rattenkrieg,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

**Comment by johanandren** [comment]
OK TO TEST

**Comment by johanandren** [comment]
PLS BUILD

File: akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowGroupedWithinSpec.scala [hunk 1]
```diff
@@ -100,6 +100,20 @@ class FlowGroupedWithinSpec extends StreamSpec with ScriptedTest {
...
 
+    "not emit empty group when finished while not being pushed" taggedAs TimingTest in {
+      val p = TestPublisher.manualProbe[Int]()
+      val c = TestSubscriber.manualProbe[immutable.Seq[Int]]()
+      Source.fromPublisher(p).groupedWithin(1000, 500.millis).to(Sink.fromSubscriber(c)).run()
+      val pSub = p.expectSubscription
+      val cSub = c.expectSubscription
+      cSub.request(1)
+      pSub.expectRequest
+      c.expectNoMsg(600.millis)
```
Comment (johanandren) [inline akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowGroupedWithinSpec.scala]: Is this delay (and the high `groupedWithin`one) really needed to exercise the bug?

**Review by johanandren — APPROVED** [review]
LGTM, thanks @Rattenkrieg ! Would be nice if testcase could be quicker though.

**Comment by johanandren** [comment]
OK TO TEST

File: akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowGroupedWithinSpec.scala (hunk 1 above)
Comment (Rattenkrieg) [inline akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowGroupedWithinSpec.scala]: I just have copy-pasted numbers from test above this one. Will try with lower values.

Comment (johanandren) [inline akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowGroupedWithinSpec.scala]: I was thinking that maybe it doesn't need the expectNoMsg at all, if it is just about completing without upstream emitting anything?

**Comment by Rattenkrieg** [comment]
@johanandren you was right we don't need `expectNoMsg` here. I have checked with broken code, test still failing without  `expectNoMsg` and with lowered timings. Now I'm holding fastest execution time record among this testsuite 🥇

**Comment by johanandren** [comment]
Excellent, thanks for taking the time to check that!

**Comment by johanandren** [comment]
OK TO TEST

**Review by patriknw — APPROVED** [review]
LGTM

**Comment by patriknw** [comment]
@Rattenkrieg Thanks for fixing.

**Comment by patriknw** [comment]
I'll backport

**Pull request merged**